### PR TITLE
Transcript UI implementation

### PR DESCRIPTION
**Pull request by Dananji**
This is how the transcript component looks with the changes in this PR:

transcript

### CODE REVIEW

File: src/components/Transcript/Transcript.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  const [transcript, setTranscript] = React.useState([]);
+  const [transcriptTitle, setTranscriptTitle] = React.useState('');
+  const [transcriptUrl, setTranscriptUrl] = React.useState('');
+
+  // React refs array for each timed text value in the transcript
+  let textRefs = React.useRef([]);
+  const transcriptContainerRef = React.useRef();
+
+  React.useEffect(() => {
+    if (transcripts && transcripts.length > 0) {
```
Comment (adamjarling) [inline src/components/Transcript/Transcript.js:16]: Might be able to short-circuit this evaluation with a `if (transcripts?.length > 0)...`

File: src/components/Transcript/Transcript.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+      autoScrollAndHighlight(73004);
+    }, 3000);
+    setTimeout(function () {
+      autoScrollAndHighlight(27531);
+    }, 6000);
+  };
+
+  const autoScrollAndHighlight = (currenttime) => {
+    let textTopOffset = 0;
+    textRefs.current.map((tr) => {
```
Comment (adamjarling) [inline src/components/Transcript/Transcript.js:51]: `map` returns an array, so if you just looking to loop through `textRefs`, might be more clear to do a `forEach` or a `for...of` or `for...in`?

File: src/components/Transcript/TranscriptMenu/TranscriptSelector.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import TranscriptDownloader from './TranscriptDownloader';
+
```
Comment (adamjarling) [inline src/components/Transcript/TranscriptMenu/TranscriptSelector.js:3]: Might want to pull in the `prop-types` package just to make it more clear to other devs, which props are expected for the component?

**Review by adamjarling — APPROVED** [review]

File: src/components/Transcript/Transcript.js (hunk 1 above)
Comment (Dananji) [reply src/components/Transcript/Transcript.js:16]: Didn't know that javascript had that :100:

File: src/components/Transcript/TranscriptMenu/TranscriptSelector.js (hunk 3 above)
Comment (Dananji) [reply src/components/Transcript/TranscriptMenu/TranscriptSelector.js:3]: Absolutely!

**Pull request merged**